Doberman/HostMonitor.py

```diff
@@ -0,0 +1,183 @@
+import Doberman
+import psutil
+import datetime
+from socket import getfqdn
+import threading
+
+dtnow = datetime.datetime.utcnow
+
+__all__ = 'HostMonitor'.split()
+
+
+class HostMonitor(Doberman.Monitor):
+    """
+    A Monitor subclass to monitor a host. Makes sure a host's sensors are always
+    running
+    """
+
+    def setup(self):
+        self.hostname = getfqdn()
+        self.kafka = self.db.get_kafka("sysmon")
+        self.sh = Doberman.utils.SignalHandler(self.logger, self.event)
+        cfg = self.db.get_host_setting()
+        self.db.set_host_setting(self.hostname, set={'status': 'online'})
+        self.last_restart_times = {}
+        swap = psutil.swap_memory()
+        self.last_swap_in = swap.sin
+        self.last_swap_out = swap.sout
+        self.sysmon_timer = int(cfg['sysmon_timer'])
+        self.disks = cfg['disks']
+        self.last_read = {}
+        self.last_write = {}
+        disk_io = psutil.disk_io_counters(True)
+        for disk in self.disks:
+            self.last_read[disk] = disk_io[disk].read_bytes
+            self.last_write[disk] = disk_io[disk].write_bytes
+        self.nics = cfg['nics']
+        net_io = psutil.net_io_counters(True)
+        self.last_recv = {}
+        self.last_sent = {}
+        for nic in self.nics:
+            self.last_recv[nic] = net_io[nic].bytes_recv
+            self.last_sent[nic] = net_io[nic].bytes_sent
+        self.register(obj=self.system_status, period=cfg['sysmon_timer'], name='sysmon')
+        self.register(obj=self.heartbeat, period=cfg['heartbeat_timer'], name='heartbeat')
+
+    def system_status(self):
```

Comment:
This function should probably be its own script that just always runs in the background, it can be a bad idea to lose information about host health just because something else in this (large and complex) package isn't behaving properly